daemon/cluster/cluster.go

```diff
@@ -625,7 +605,16 @@ func (c *Cluster) Update(version uint64, spec types.Spec, flags types.UpdateFlag
 		return err
 	}
 
-	swarmSpec, err := convert.SwarmSpecToGRPC(spec)
+	// In update, client should specify the desired state, which includes
+	// Name and Labels. So we need to initialize the name and labels here
+	// based on the value in spec.
+	// It is possible that empty Labels be specified (unset in update).
+	swarmSpec, err := convert.SwarmSpecToGRPC(spec, swarmapi.ClusterSpec{
+		Annotations: swarmapi.Annotations{
+			Name:   spec.Name,
+			Labels: spec.Labels,
+		},
+	})
```

Comment:
@aaronlehmann The pull request has been updated to address the above mentioned. Please let me know if there are any issues.